source/common/network/dns_impl.cc

```diff
@@ -209,7 +209,14 @@ void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, i
 
   if (!completed_ && fallback_if_failed_) {
     fallback_if_failed_ = false;
-    getAddrInfo(AF_INET);
+
+    if (dns_lookup_family_ == DnsLookupFamily::Auto) {
+      getAddrInfo(AF_INET);
+    } else {
+      ASSERT(dns_lookup_family_ == DnsLookupFamily::V4Preferred);
```

Comment:
Is this assert valid? Can't you get into this branch from `::V6Only`?

Reply:
no, ::V6Only sets `fallback_if_failed_` to false